### DISCUSSION TITLE
request.ip with trustProxy as true

### DISCUSSION THREAD

**Discussion by outbackStack** [fastify/fastify#5304] [Q&A] [answered]
I've been using trustProxy set to `true` to get the user's ip address without any issues but about a week ago it's giving me `127.0.0.1` instead. Is anyone having the same issue?

```
const fastify = Fastify({
  trustProxy: true
});
``

fastify.get('/', async (request, reply) {
 console.log(request.ip);
`127.0.0.1`

return reply
})

**Comment by metcoder95** [comment]
Have you checked your `request.ips` values?

**Comment by outbackStack** [comment]
I have, it's just in an array instead `["127.0.0.1"]`.

I'm running with proxy_pass using NGINX...maybe that's the reason but it was working before. It's weird to me as I haven't updated anything.

```
upstream nodeapp1 {
        hash $binary_remote_addr consistent;
        zone upstreams 64K;
	server 127.0.0.1:xxxx;

	keepalive 2;
}
```

**Reply by metcoder95** [reply, marked as answer]
Hmm, by any chance have you debug what are the values that you received through `x-forwarded-host` or `x-forwarded-proto`? Have you recently updated your `nginx` version or so?
Maybe you should also verify your are mapping the headers correctly, see [this](https://www.nginx.com/resources/wiki/start/topics/examples/forwarded/) as reference

**Reply by outbackStack** [reply]
thank you @metcoder95 
Setting the `proxy_set_header` in the `proxy_pass block`, does the trick :-)
